lib/_tls_common.js

```diff
@@ -153,6 +153,38 @@ exports.createSecureContext = function createSecureContext(options) {
     }
   }
 
+  const engineKey = options.engineKey;
+  const privateKeyEngine = options.privateKeyEngine;
+  if (engineKey && privateKeyEngine) {
```

Comment:
Node.js has traditionally been lazy when it comes to input validation, but new APIs should be stricter: Use `!== undefined` or something similar to not ignore inputs such as `engineKey: 0`.

Reply:
Also, I don't know if it would make sense to validate types even if only one option is given? Or to even throw if only one is given? Is there any use case for only specifying one of these options?

Reply:
> Is there any use case for only specifying one of these options?

@tniessen From the OpenSSL point of view, you can extract multiple keys from one engine, but since we only have one key in the SecureContext, they should probably go together. On the other hand, for client certificate we have a separate option to set the engine, and then to set the certificate, but there is a OpenSSL API call for setting certificate engine, while there is nothing to set "key engine".

Reply:
I agree, but I would prefer to throw when only one of the options is given. (Currently, you are ignoring the options if only one is used.) If there is no legitimate use case for only specifying one of the two options, then it is most likely an error or a mistake, and we shouldn't silently ignore that.

Reply:
I'm not sure this comment was addressed, though the code was outdated, since if only one of engine or keyId is provided, they are  both ignored.